src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useRef, useState } from "react";
+import ArrowDownIcon from "../icons/ArrowDownIcon";
+import styles from "./Dropdown.module.scss";
+
+export type Option = {
+  key: string;
+  value: string;
+};
+
+export type DropdownProps = {
+  /** Массив возможных вариантов для выбора */
+  options: Option[];
+  /** Текущий индекс, может быть пустым */
+  keySelected: string;
+  /** Callback, вызываемый при выборе варианта */
+  onChange: (key: string) => void;
+};
+
+const Dropdown: React.FC<DropdownProps> = ({
+  options,
+  keySelected,
+  onChange,
+}: DropdownProps) => {
+  let [isExpanded, Toggle] = useState(false);
+  let [index, SetIndex] = useState(keySelected);
+
+  const newRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    document.addEventListener("mousedown", handleOutsideClick);
+    return () => {
+      document.removeEventListener("mousedown", handleOutsideClick);
+    };
+  });
+  const handleOutsideClick = (e: any) => {
+    if (newRef.current && !newRef.current.contains(e.target)) {
+      Toggle(false);
+    }
+  };
+
+  const handleOnListClicked = (key: string) => {
+    {
+      Toggle(false);
+      //only 1 category
+      // if selected, then unselect
+      if (key == index) {
+        onChange(key); //callback
+        SetIndex("");
+      } else {
+        onChange(key); //callback
+        SetIndex(key);
+      }
+    }
+  };
+
+  const listItems = options.map((elm) => (
+    <div
+      key={elm.key}
+      onClick={() => {
+        handleOnListClicked(elm.key);
+      }}
+      className={elm.key == index ? styles.itemSelected : ""}
```

Comment:
```suggestion
      className={elm.key === index ? styles.itemSelected : ""}
```
Желательно везде использовать оператор строгого равенства

Reply:
поправил везде, где нашел.